### PR TITLE
Moved JSON summary prompt to mention score is an integer

### PR DESCRIPTION
**Pull request by jamesbraza**
We keep seeing (both internally and in https://github.com/Future-House/paper-qa/issues/747) the JSON context answers with float scores (e.g. 7.5).

I realized the non-JSON summary prompt explicitly states "integer":

> provide an integer score from 1-10

But the JSON summary prompt does not:

> relevance of `summary` to answer question (out of 10)

This PR standardizes the two prompts to mention integer scores between 1-10.

It also cleans up the `extract_score` function a bit and adds docs to it

### CODE REVIEW

File: paperqa/prompts.py [hunk 1]
```diff
@@ -78,7 +78,7 @@
   "relevance_score": "..."
 }}
 
-where `summary` is relevant information from text - {summary_length} words and `relevance_score` is the relevance of `summary` to answer question (out of 10).
+where `summary` is relevant information from text - {summary_length} words and `relevance_score` is an integer score from 1-10 for the relevance of `summary` to answer question.
```
Comment (whitead) [inline paperqa/prompts.py:81]:
```suggestion
where `summary` is relevant information from the text - {summary_length} words. `relevance_score` is an integer 1-10 for the relevance of `summary` to answer the question.
```

File: paperqa/utils.py [hunk 2]
```diff
@@ -130,12 +130,19 @@ def strip_citations(text: str) -> str:
...
+    """
+    # check for N/A, NA, not applicable, not relevant
     last_line = text.split("\n")[-1]
-    if "N/A" in last_line or "n/a" in last_line or "NA" in last_line:
-        return 0
-    # check for not applicable, not relevant in summary
-    if "not applicable" in text.lower() or "not relevant" in text.lower():
+    if (
+        "n/a" in last_line.lower()
+        or "NA" in last_line
```
Comment (whitead) [inline paperqa/utils.py:142]: This looks like a bug. For example:

"The gene NAR is explicitly mentioned in the text, thus the relevance score is 10/10"

**Review by mskarlin — APPROVED** [review]

File: paperqa/utils.py (hunk 2 above)
Comment (jamesbraza) [reply paperqa/utils.py]: Fwiw, the `"NA"` check was previously there before this PR, this change was just a refactor for readability.

I agree with your comment through, so I removed the `"NA"` check and added this comment:

```python
# Don't check for NA, as there can genes containing NA
```

File: paperqa/prompts.py (hunk 1 above)
Comment (whitead) [reply paperqa/prompts.py]: Just want to clean up grammar here, as long as we're editing

**Review by whitead — APPROVED** [review]

**Pull request merged**